From e6461d63c1a99a1d797f0bef77722cc94530d269 Mon Sep 17 00:00:00 2001
From: Leonard Crestez <leonard.crestez@nxp.com>
Date: Mon, 18 Jun 2018 21:03:48 +0300
Subject: [PATCH 3988/5242] MLK-18628-1 ARM: dts: imx7d: Add pwm polarity
 specifier for backlight

commit  7b67689edea54241062c85ae4b41d2c660029c2c from
https://source.codeaurora.org/external/imx/linux-imx.git

Upstream modified imx7s.dtsi pwm to have #pwm-cells = <3> so now our
out-of-tree boards need to be modified to use 3 cells.

See commit 9be48d2d9881 ("ARM: dts: imx7: use 3 PWM cells")

Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Review-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7d-12x12-ddr3-arm2.dts   |    2 +-
 arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts |    2 +-
 arch/arm/boot/dts/imx7d-sdb.dts               |    2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/imx7d-12x12-ddr3-arm2.dts b/arch/arm/boot/dts/imx7d-12x12-ddr3-arm2.dts
index 1477968..8455815 100644
--- a/arch/arm/boot/dts/imx7d-12x12-ddr3-arm2.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-ddr3-arm2.dts
@@ -17,7 +17,7 @@
 
 	backlight {
 		compatible = "pwm-backlight";
-		pwms = <&pwm1 0 5000000>;
+		pwms = <&pwm1 0 5000000 0>;
 		brightness-levels = <0 4 8 16 32 64 128 255>;
 		default-brightness-level = <6>;
 		status = "okay";
diff --git a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
index 1d12619..f8f1eed 100644
--- a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
@@ -16,7 +16,7 @@
 
 	backlight {
 		compatible = "pwm-backlight";
-		pwms = <&pwm1 0 5000000>;
+		pwms = <&pwm1 0 5000000 0>;
 		brightness-levels = <0 4 8 16 32 64 128 255>;
 		default-brightness-level = <6>;
 		status = "okay";
diff --git a/arch/arm/boot/dts/imx7d-sdb.dts b/arch/arm/boot/dts/imx7d-sdb.dts
index 4222b5f..1c6e2b2 100644
--- a/arch/arm/boot/dts/imx7d-sdb.dts
+++ b/arch/arm/boot/dts/imx7d-sdb.dts
@@ -130,7 +130,7 @@
 
 	backlight {
 		compatible = "pwm-backlight";
-		pwms = <&pwm1 0 5000000>;
+		pwms = <&pwm1 0 5000000 0>;
 		brightness-levels = <0 4 8 16 32 64 128 255>;
 		default-brightness-level = <6>;
 		status = "okay";
-- 
1.7.9.5

